oak/server/node_thread.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OAK_SERVER_PSEUDO_NODE_H_
+#define OAK_SERVER_PSEUDO_NODE_H_
+
+#include <atomic>
+#include <memory>
+#include <thread>
+
+namespace oak {
+
+// This class represents a node or pseudo-node that executes as a distinct
+// thread.
+class NodeThread {
+ public:
+  NodeThread(const std::string& name) : name_(name), termination_pending_(false) {}
```

Comment:
What is `name` used for? Debugging / log messages? Could you clarify in a comment? If that's the case probably a generic name would not be enough, presumably you would want an ID of some sort, so that log entries from multiple nodes of the same type can be distinguished. And probably it would be useful for various other things to be able to refer to nodes by some ID.